bookstore/handlers.py

```diff
@@ -35,41 +33,44 @@ class BookstorePublishHandler(APIHandler):
     def __init__(self, *args, **kwargs):
         super(APIHandler, self).__init__(*args, **kwargs)
         # create an easy helper to get at our bookstore settings quickly
-        self.bookstore_settings = BookstoreSettings(
-            config=self.settings['config']['BookstoreSettings']
-        )
-
-        self.fs = s3fs.S3FileSystem(
-            key=self.bookstore_settings.s3_access_key_id,
-            secret=self.bookstore_settings.s3_secret_access_key,
-            client_kwargs={
-                "endpoint_url": self.bookstore_settings.s3_endpoint_url,
-                "region_name": self.bookstore_settings.s3_region_name,
-            },
-            config_kwargs={},
-            s3_additional_kwargs={},
-        )
+        self.bookstore_settings = BookstoreSettings(config=self.config)
 
-    @property
-    def bucket(self):
-        return self.bookstore_settings.s3_bucket
+        self.session = aiobotocore.get_session()
 
-    @property
-    def prefix(self):
-        return self.bookstore_settings.published_prefix
+    # @property
+    # def bucket(self):
+    #     return self.bookstore_settings.s3_bucket
+    #
+    # @property
+    # def prefix(self):
+    #     return self.bookstore_settings.published_prefix
 
-    def s3_path(self, path):
-        """compute the s3 path based on the bucket, prefix, and the path to the notebook"""
-        return s3_path(self.bucket, self.prefix, path)
+    # def s3_path(self, path):
+    #     """compute the s3 path based on the bucket, prefix, and the path to the notebook"""
+    #     return s3_path(self.bucket, self.prefix, path)
 
-    def _publish(self, model, path):
+    async def _publish(self, model, path):
         if model['type'] != 'notebook':
             raise web.HTTPError(400, "bookstore only publishes notebooks")
         content = model['content']
 
-        full_s3_path = self.s3_path(path)
-
-        self.log.info("Publishing to %s", s3_display_path(self.bucket, self.prefix, path))
+        full_s3_path = s3_path(self.bookstore_settings.s3_bucket, self.bookstore_settings.published_prefix, path)
+        file_key = s3_key(self.bookstore_settings.published_prefix, path)
+
+        self.log.info("Publishing to %s", s3_display_path(self.bookstore_settings.s3_bucket,
+                                                          self.bookstore_settings.published_prefix,
+                                                          path))
+        async with self.session.create_client('s3',
```

Comment:
Let's try to use Black's code style when it comes to line continuations since it's a bit easier to scan: https://github.com/ambv/black#how-black-wraps-lines

```
async with self.session.create_client(
    's3',
     aws_secret_access_key=self.bookstore_settings.s3_secret_access_key,
     aws_access_key_id=self.bookstore_settings.s3_access_key_id,
     endpoint_url=self.bookstore_settings.s3_endpoint_url,
     region_name=self.bookstore_settings.s3_region_name, ) as client:
```